### PR TITLE
[avro] Add documentation for the Avro format.

### PR DESCRIPTION
**Pull request by ryzhyk**

### CODE REVIEW

File: crates/feldera-types/src/format/avro.rs [hunk 1]
```diff
@@ -62,33 +62,46 @@ impl AvroUpdateFormat {
...
+    /// * **Output connector**: When non-empty, the connector will
+    ///   post the schema to the registry and embed the schema id returned
+    ///   by the registry in Avro messages.  Otherwise, schema id 0 is used.
     #[serde(default)]
     pub registry_urls: Vec<String>,
+
     /// Custom headers that will be added to every call to the schema registry.
     ///
+    /// This property is only applicable to output connectors.
+    ///
```
Comment (gz) [inline crates/feldera-types/src/format/avro.rs:79]:
```suggestion
```

File: crates/feldera-types/src/format/avro.rs [hunk 2]
```diff
@@ -106,20 +119,21 @@ pub struct AvroParserConfig {
...
-    /// Avro schema used to encode input records.
+    /// Avro schema used to encode all records in this stream, specified as a JSON-encoded string.
     ///
-    /// Specified as a string containing schema definition in JSON format.
-    /// This schema must match precisely the SQL view definition, including
-    /// nullability of columns.
+    /// When this property is set, the connector uses the provided schema instead of
+    /// retrieving the schema from the schema registry. This setting is mutually exclusive
+    /// with `registry_urls`.
     pub schema: Option<String>,
```
Comment (gz) [inline crates/feldera-types/src/format/avro.rs:127]: don't we usually use Object for json stuff?

File: docs/formats/avro.md [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+
+:::note
+This page describes configuration options specific to the Avro data format.
+See [top-level connector documentation](/docs/connectors/) for general information
+about configuring input and output connectors.
+:::
+
+Feldera supports sending and receiving data in the Avro format. While Avro-encoded messages are commonly
+transmitted over Kafka, any other supported [transport](/docs/connectors/) can be used.
+Avro is a strongly-typed format that requires a shared **schema** between the sender and receiver for successfu
```
Comment (gz) [inline docs/formats/avro.md:11]:
```suggestion
Avro is a strongly-typed format that requires a shared **schema** between the sender and receiver for successful
```

File: docs/formats/avro.md [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+* **Raw** - every message contains a single Avro-encoded record that represents a row in a SQL
+  table or view. This format does not carry any additional metadata that can be used to distinguish
+  inserts and deletes.  It is therefore only suitable for representing inserts and upserts, but not
+  deletions.
+
+* **Debezium** (input only) - used to synchronize a Feldera table with an external database using
+  [Debezium](https://debezium.io/).
+
+* **Confluent JDBC** (output only) - used to send incremental changes computed by Feldera
+  to an external database using [Confluent JDBC connector](https://docs.confluent.io/kafka-connectors/jdbc/current/sink-connector/)
```
Comment (gz) [inline docs/formats/avro.md:34]:
```suggestion
  to an external database using the [Confluent JDBC connector](https://docs.confluent.io/kafka-connectors/jdbc/current/sink-connector/)
```

File: docs/formats/avro.md [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+* **Sender**: Before sending the first message with a new schema, the sender registers the schema
+  in the schema registry.
+* **Receiver**: Upon receiving a message, the receiver retrieves the associated schema from the
+  registry based on the schema identifier before decoding the message.
+
+We support several Avro-based formats:
+
+* **Raw** - every message contains a single Avro-encoded record that represents a row in a SQL
+  table or view. This format does not carry any additional metadata that can be used to distinguish
+  inserts and deletes.  It is therefore only suitable for representing inserts and upserts, but not
```
Comment (gz) [inline docs/formats/avro.md:27]: add link to where we explain how upserts work in feldera (if we have one?)

File: docs/formats/avro.md [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+* The Avro schema must be of type `record`.
+
+* For every non-nullable column in the table, a field with the same name and a compatible type must be present in the Avro schema
+  Note that the Avro schema is allowed to contain fields that don't exist in the SQL table.  Such fields are ignored by the parser.
+  Conversely, the SQL table can contain **nullable** columns that are not present in the schema.  Such columns will
+  be set to `NULL` during deserialization.
+
+A SQL column and a field in the Avro schema are compatible if:
+
+* If the Avro field is nullable, then the SQL column must be nullable too (however, a non-nullable
```
Comment (gz) [inline docs/formats/avro.md:69]:
```suggestion
* The Avro field is nullable, then the SQL column must be nullable too (however, a non-nullable
```

File: docs/formats/avro.md [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+  field can be deserialized into eother nullable or non-nullable column).
+
+* SQL column type and Acro field type must match according to the following table.
```
Comment (gz) [inline docs/formats/avro.md:72]:
```suggestion
* The SQL column type and Acro field type must match according to the following table.
```

File: docs/formats/avro.md [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+| `TIME`                        | `long` or `int`| logical type must be set to `time-millis` or `time-micros`          |
+| `TIMESTAMP`                   | `long`         | logical type must be set to `timestamp-millis` or `timestamp-micros`|
+| `ARRAY`                       | `array`        | Avro and SQL array element schemas must match                       |
+| `MAP`                         | `map`          | SQL map keys must be of type `CHAR` or `VARCHAR`; Avro and SQL value schemas must match|
+| user-defined types            | `record`       | Avro record schema must match SQL user-defined type definition according to the same schema compatibility rules as for SQL tables|
+
+### Configuration
+
+The following properties can be used to configure the Avro parser. All of these properties are optional.  However,
+exactly one of `registry_urls` and `schema` properties must be specified.
```
Comment (gz) [inline docs/formats/avro.md:94]:
```suggestion
either `registry_urls` or `schema` properties must be specified.
```

File: docs/formats/avro.md [hunk 9]
```diff
@@ -0,0 +1,257 @@
...
+| `registry_proxy`              | string          | | Proxy that will be used to access the schema registry. Requires `registry_urls` to be set.|
+| `registry_timeout_secs`       | string          | | Timeout in seconds used to connect to the registry. Requires `registry_urls` to be set.|
+| `registry_username`           | string          | | Username used to authenticate with the registry.Requires `registry_urls` to be set. This option is mutually exclusive with token-based authentication (see `registry_authorization_token`).|
+| `registry_password`           | string          | | Password used to authenticate with the registry. Requires `registry_urls` to be set.|
+| `registry_authorization_token`| string          | | Token used to authenticate with the registry. Requires `registry_urls` to be set. This option is mutually exclusive with password-based authentication (see `registry_username` and `registry_password`).|
+
+### Examples
+
+**Example 1**: Configure Avro parser to receive raw Avro records without embedded schema ids using a static user-provided schema.
```
Comment (gz) [inline docs/formats/avro.md:110]:
```suggestion
* Configure the Avro parser to receive raw Avro records without embedded schema ids, using a static user-provided schema.
```

File: docs/formats/avro.md [hunk 10]
```diff
@@ -0,0 +1,257 @@
...
+      "config": {
+        "schema": "{\"type\":\"record\",\"name\":\"ExampleSchema\",\"fields\":[{\"name\":\"id\",\"type\":\"int\"},{\"name\":\"ts\",\"type\":[\"null\",{\"type\":\"long\",\"logicalType\":\"timestamp-micros\"}]}]}",
+        "skip_schema_id": true,
+        "update_format": "raw"
+      }
+    }
+}]');
+```
+
+**Example 2**: Configure Avro parser to ingest data change events from Debezium
```
Comment (gz) [inline docs/formats/avro.md:137]:
```suggestion
* Configure the Avro parser to ingest data change events from Debezium
```

File: docs/formats/avro.md [hunk 11]
```diff
@@ -0,0 +1,257 @@
...
+```
+
+## Avro output
+
+### Schema management
+
+The Avro encoder generates an Avro schema, which the consumer can use to decode messages produced
+by the encoder. For message formats that include key and value components with different schemas,
+e.g., the Confluent JDBC connector format, the encoder generates both key and value schemas.
+If the connector configuration specifies a schema registry, the encoder publishes both
```
Comment (gz) [inline docs/formats/avro.md:172]: maybe add link to the relevant example

File: docs/formats/avro.md [hunk 12]
```diff
@@ -0,0 +1,257 @@
...
+key and value schemas in the registry.
+
+The encoder supports an alternative workflow where users provide the schema as a JSON string
```
Comment (gz) [inline docs/formats/avro.md:175]: another link to example

File: docs/formats/avro.md [hunk 13]
```diff
@@ -0,0 +1,257 @@
...
+| `registry_proxy`              | string          | | Proxy that will be used to access the schema registry. Requires `registry_urls` to be set.|
+| `registry_timeout_secs`       | string          | | Timeout in seconds used to connect to the registry. Requires `registry_urls` to be set.|
+| `registry_username`           | string          | | Username used to authenticate with the registry.Requires `registry_urls` to be set. This option is mutually exclusive with token-based authentication (see `registry_authorization_token`).|
+| `registry_password`           | string          | | Password used to authenticate with the registry. Requires `registry_urls` to be set.|
+| `registry_authorization_token`| string          | | Token used to authenticate with the registry. Requires `registry_urls` to be set. This option is mutually exclusive with password-based authentication (see `registry_username` and `registry_password`).|
+
+### Examples
+
+**Example 1**: Configure Avro encoder to send raw Avro records using a static user-provided schema.  The configuration does not
```
Comment (gz) [inline docs/formats/avro.md:203]:
```suggestion
* Configure the Avro encoder to send raw Avro records using a static user-provided schema.  The configuration does not
```

File: docs/formats/avro.md [hunk 14]
```diff
@@ -0,0 +1,257 @@
...
+        "schema": "{\"type\":\"record\",\"name\":\"ExampleSchema\",\"fields\":[{\"name\":\"id\",\"type\":\"int\"},{\"name\":\"ts\",\"type\":[\"null\",{\"type\":\"long\",\"logicalType\":\"timestamp-micros\"}]}]}"
+      }
+    }
+  }]'
+)
+AS
+    SELECT * from my_table;
+```
+
+**Example 2**: Configure Avro encoder to output changes in the format expected by the
```
Comment (gz) [inline docs/formats/avro.md:230]:
```suggestion
* Configure the Avro encoder to output changes in the format expected by the
```

**Review by gz — APPROVED** [review]

File: crates/feldera-types/src/format/avro.rs (hunk 2 above)
Comment (ryzhyk) [reply crates/feldera-types/src/format/avro.rs:127]: Not sure what's an object.  We could try parsing this into `serde_json::Value`, but this is an escaped string, not a regular piece of JSON, so not sure how to do this exactly, plus the library function for deserializing schemas works with strings.

**Pull request merged**